Source/missiles.cpp

```diff
@@ -1429,19 +1429,27 @@ void AddLightningArrow(Missile &missile, const AddMissileParameter &parameter)
 	UpdateMissileVelocity(missile, dst, 32);
 	missile._miAnimFrame = GenerateRnd(8) + 1;
 	missile._mirange = 255;
-	if (missile._misource < 0) {
+	switch (missile.sourceType()) {
+	case MissileSource::Trap:
 		missile.var1 = missile.position.start.x;
 		missile.var2 = missile.position.start.y;
-	} else {
-		missile.var1 = Players[missile._misource].position.tile.x;
-		missile.var2 = Players[missile._misource].position.tile.y;
+		break;
+	case MissileSource::Player: {
+		Player &player = *missile.sourcePlayer();
+		missile.var1 = player.position.tile.x;
+		missile.var2 = player.position.tile.y;
+	} break;
+	case MissileSource::Monster: {
+		assert(missile.sourceType() != MissileSource::Monster);

```

Comment:
This is a bit odd, perhaps just `app_fatal("Invalid missile source")`

Reply:
There was already some discussion about that pattern on https://github.com/diasurgical/devilutionX/pull/5098#discussion_r928109064 https://github.com/diasurgical/devilutionX/pull/5098#discussion_r928305171

I followed the pattern from that one

Reply:
The app_fatal() is also fine and is how it's normally done, except it will change it to also crash in production.